File: frontend/app/src/main/java/com/example/caffwebshop/model/CaffItemPublic.kt
package com.example.caffwebshop.model

import com.google.gson.annotations.SerializedName

data class CaffItemPublic (
    @SerializedName("id")
    val id: Int,
    @SerializedName("title")
    val title: String,
    @SerializedName("tags")
    val tags: List<String>,
    @SerializedName("ownerId")
    val ownerId: Int,
    @SerializedName("uploadedAt")
    val uploadedAt: String,
    @SerializedName("owner")
    val owner: UserPublic
)

File: backend/CaffShop/Models/CaffItem/CaffItemPublic.cs
using System;
using System.Collections.Generic;
using CaffShop.Models.User;

namespace CaffShop.Models.CaffItem
{
    public class CaffItemPublic
    {
        public long Id { get; set; }
        public string Title { get; set; }
        // ReSharper disable once CollectionNeverUpdated.Global
        public List<string> Tags { get; set; }
        public long OwnerId { get; set; }
        public DateTime UploadedAt { get; set; }
        public UserPublic Owner { get; set; }
    }
}